tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -317,7 +317,16 @@ public function filterResponseByNotificationSubjectAndResource(string $subject,
 	 * @return void
 	 */
 	public function userShouldGetANotificationWithMessage(string $user, string $subject, TableNode $table):void {
-		$this->userListAllNotifications($user);
+		$count = 0;
+		// sometimes the test might try to get notification before the notification is created by the server
+		// in order to prevent test from failing because of that list the notifications again
+		while (!isset($this->filterResponseAccordingToNotificationSubject($subject)->message) && $count <= 5) {
+			\sleep(1);
+			$this->featureContext->setResponse(null);
+			$this->userListAllNotifications($user);
+			$this->featureContext->theHTTPStatusCodeShouldBe(200);
+			++$count;
+		}
 		$actualMessage = str_replace(["\r", "\n"], " ", $this->filterResponseAccordingToNotificationSubject($subject)->message);
```

Comment:
If the loop finishes because `$count` is greater than 5, and the notification has not come, then this line will "crash" when it tries to do string replace on a non-existent message.
Before this line, we could Assert that the message is set, and if not the Assert can have a nice message saying that the notification was not found after trying for 5 seconds.